src/app/fdctl/run/tiles/fd_poh.c

```diff
@@ -962,6 +1069,39 @@ fd_ext_poh_reset( ulong         completed_bank_slot, /* The slot that successful
   ctx->next_leader_slot = next_leader_slot( ctx );
   FD_LOG_INFO(( "fd_ext_poh_reset(slot=%lu,next_leader_slot=%lu)", ctx->reset_slot, ctx->next_leader_slot ));
 
+  if( FD_UNLIKELY( ctx->slot>=ctx->next_leader_slot ) ) {
+    /* We are leader after the reset... two cases: */
+    if( FD_LIKELY( ctx->slot==slot_before_reset ) ) {
+      /* 1. We are reset onto the same slot we are already leader on.
+            This is a common case when we have two leader slots in a
+            row, replay stage will reset us to our own slot.  No need to
+            do anything here, we already sent a SLOT_START. */
+      FD_TEST( leader_before_reset );
+    } else {
+      /* 2. We are reset onto a different slot. If we were leader
+            before, we should first end that slot, then begin the new
+            one if we are newly leader now. */
+      if( FD_LIKELY( leader_before_reset ) ) {
+        uchar data[ 16 ];
+        FD_STORE( ulong, data, slot_before_reset );
+        FD_STORE( ulong, data+8UL, ctx->cus_used );
+        poh_link_publish( &poh_plugin, FD_PLUGIN_MSG_SLOT_END, data, 16UL );
```

Comment:
I think something like
```suggestion
        struct {
          ulong slot;
          ulong cus;
          } end = { slot_before_reset, ctx->cus_used };
          poh_link_publish( &poh_plugin, FD_PLUGIN_MSG_SLOT_END, (char const *)&end, sizeof(end) );
```


```suggestion
        uchar data[ 16 ];
        FD_STORE( ulong, data, slot_before_reset );
        FD_STORE( ulong, data+8UL, ctx->cus_used );
        poh_link_publish( &poh_plugin, FD_PLUGIN_MSG_SLOT_END, data, 16UL );
```

Reply:
But I guess perhaps that's a violation of strict aliasing? So then the alternative would be 
`union {
  uchar bytes[16];
  struct {
    ulong slot;
    ulong cus;
  } v;
} end = { .v = { slot_before_reset, ctx->cus_used } };
poh_link_publish( &poh_plugin, FD_PLUGIN_MSG_SLOT_END, end.bytes, sizeof(end) );`
It's not clear to me this is less gross, but I'll leave it for your consideration